src/convert.js

```diff
@@ -128,72 +129,79 @@ function convertCamelCaseToKebabCase(string) {
 }
 
 function exportToJS(
-  resolvedTokensByCategory,
+  tokensByCategory,
   staticInterfaces,
   outDir,
   outFileName,
   interfacesOutFileName
 ) {
-  const jSExport = formatToJS(resolvedTokensByCategory);
-  const typesExport = generateTSInterfaces(
-    resolvedTokensByCategory,
-    staticInterfaces
-  );
+  const jSExport = formatToJS(tokensByCategory);
+  const typesExport = generateTSInterfaces(tokensByCategory, staticInterfaces);
   exportFile(`${outDir}`, `${outFileName}.js`, jSExport);
   exportFile(`${outDir}`, interfacesOutFileName, typesExport);
 }
 
-function formatToJS(resolvedTokensByCategory) {
-  let jSExport = {};
-  resolvedTokensByCategory.forEach(category => {
-    switch (category.category) {
-      case "colors":
-        jSExport.colors = formatColorsToJS(category.tokens);
-        break;
-      case "typography":
-        jSExport.typograhphy = formatTypographyToJS(category.tokens);
-        break;
-      case "spacing":
-        jSExport.spacing = formatSpacingToJS(category.tokens);
-        break;
-      default:
-        console.warn("Unrecognized category type");
-    }
-  });
-  return `module.exports = ` + JSON.stringify(jSExport);
+function formatToJS(tokensByCategory) {
+  const jSExport = Object.assign(
+    {},
+    ...tokensByCategory.reduce((resultArray, category) => {
+      switch (category.category) {
+        case "colors":
+          resultArray.push(formatColorsToJS(category.tokens));
+          break;
+        case "typography":
+          resultArray.push(formatTypographyToJS(category.tokens));
+          break;
+        case "spacing":
+          resultArray.push(formatSpacingToJS(category.tokens));
+          break;
+        default:
+          console.warn("Unrecognized category type");
+      }
+      return resultArray;
+    }, [])
+  );
+  return "export const tokens = " + JSON.stringify(jSExport);
 }
 
 function formatColorsToJS(tokens) {
-  let colors = {};
-  tokens.forEach(token => {
-    colors[token.name] = {
-      hsl: `hsl(${token.value.h}, ${token.value.s}%, ${token.value.l}%)`,
-      h: token.value.h,
-      s: token.value.s,
-      l: token.value.l
-    };
-  });
-  return colors;
+  return Object.assign(
```

Comment:
Assign not needed when {...tokens.map()} is enough

Reply:
Assign is actually required to compose the new object from the contents of the objects in the array rather than creating an object from with array index keys and nested objects with the actual key.